retriever/lib/provenance.py

```diff
@@ -0,0 +1,247 @@
+    if dataset.name not in os.listdir(raw_dir):
+        engine = choose_engine({"command": "download", "path": "./", "sub_dir": ""})
+        dataset.download(engine=engine, debug=quiet)
+        data_exists = True
+
+    elif dataset.name in os.listdir(raw_dir):
+        data_exists = True
+
+    if data_exists:
+        for root, _, files in os.walk(os.path.join(raw_dir, dataset.name)):
+            for file in files:
+                paths_to_zip["raw_data"].append(os.path.join(root, file))
+
+        info = commit_info_for_commit(dataset, commit_message=commit_message)
+        zip_file_name = "{}-{}{}.zip".format(
+            dataset.name, info["md5_dataset"][:3], info["md5_script"][:3]
+        )
+
+        zip_file_path = os.path.join(path, zip_file_name)
+        with ZipFile(zip_file_path, "w") as zipped:
+            zipped.write(
+                paths_to_zip["script"],
+                os.path.join("script", os.path.basename(paths_to_zip["script"])),
+            )
+            for data_file in paths_to_zip["raw_data"]:
+                zipped.write(data_file, data_file.replace(raw_dir, ""))
+            with open("metadata.json", "w") as json_file:
+                json.dump(info, json_file, sort_keys=True, indent=4)
+            zipped.write(os.path.abspath(json_file.name), "metadata.json")
+            os.remove("metadata.json")
+
+
+def commit(dataset, commit_message='', path=None, quiet=False):
+    """
+    Commit dataset to a zipped file.
+    """
+    if isinstance(dataset, str):
+        # if dataset is not a dataset script object find the right script
+        dataset = [script for script in datasets()['offline'] if script.name == dataset][0]
+    dataset_provenance_path = None if path else os.path.join(PROVENANCE_DIR, dataset.name)
+    if not path and not os.path.exists(dataset_provenance_path):
+        os.makedirs(dataset_provenance_path)
+    path = path if path else dataset_provenance_path
+    if not quiet:
+        print("Committing dataset {}".format(dataset.name))
+    try:
+        commit_writer(dataset=dataset, commit_message=commit_message, path=path, quiet=quiet)
+        if not quiet:
+            print("Successfully committed.")
+    except Exception as e:
+        print(e)
+        print("Dataset could not be committed.")
+        return
+
+
+def get_metadata(path_to_archive):
+    """
+    Returns a dictionary after reading metadata.json file of a committed dataset
+    """
+    with ZipFile(os.path.normpath(path_to_archive), 'r') as archive:
+        try:
+            metadata = json.loads(archive.read('metadata.json').decode('utf-8'))
+        except Exception as e:
+            print(e)
+            return
+    return metadata
+
+
+def commit_info_for_installation(metadata_info):
+    """
+    Returns a dictionary with commit info and changes in old and current environment
+    """
+    info = {'commit_message': metadata_info['commit_message'], 'time': metadata_info['time'], 'package_not_found': {},
```

Comment:
Refactor the line length in this file, there are some other areas where we could create shorter lines.